frontend/src/summary.pug

```diff
@@ -90,34 +90,38 @@
       .error-message-box__failed-repo--reason(v-else) {{ errorBlock.errorMessage }}
   #summary-charts
     .summary-charts(v-for="(repo, i) in filteredRepos")
-      .summary-charts__title(v-if="filterGroupSelection === 'groupByRepos'")
-        .summary-charts__title--index {{ i+1 }}
-        .summary-charts__title--repo {{ repo[0].repoName }}
-        .summary-charts__title--contribution.mini-font(
-          title="Total contribution of group"
-        ) [{{ getGroupTotalContribution(repo) }} lines]
-        template(v-if="isMergeGroup")
-          a(
-            v-bind:href="getRepoLink(repo[0])", target="_blank",
-            title="click to view group's repo"
-          )
-            i.icon-button.fab.fa-github
-        .summary-charts__title--percentile.mini-font {{ getPercentile(i) }} %
       .summary-charts__title(
-        v-if="filterGroupSelection === 'groupByAuthors'",
-        v-bind:class="{ warn: repo[0].name === '-' }"
+        v-if="filterGroupSelection === 'groupByRepos' || filterGroupSelection === 'groupByAuthors'"
       )
         .summary-charts__title--index {{ i+1 }}
-        .summary-charts__title--repo {{ repo[0].displayName }} ({{ repo[0].name }})
+        .summary-charts__title--repo
+          template(v-if="filterGroupSelection === 'groupByRepos'") {{ repo[0].repoName }}
+          template(
+            v-else-if="filterGroupSelection === 'groupByAuthors'",
+            v-bind:class="{ warn: repo[0].name === '-' }"
+          ) {{ repo[0].displayName }} ({{ repo[0].name }})
         .summary-charts__title--contribution.mini-font(
           title="Total contribution of group"
         ) [{{ getGroupTotalContribution(repo) }} lines]
         template(v-if="isMergeGroup")
+          template(v-if="filterGroupSelection === 'groupByRepos'")
+            a(
+              v-bind:href="getRepoLink(repo[0])", target="_blank",
+              title="click to view group's repo"
+            )
+              i.icon-button.fab.fa-github
+          template(v-else-if="filterGroupSelection === 'groupByAuthors'")
+            a(
+              v-bind:href="getAuthorProfileLink(repo[0].name)", target="_blank",
+              title="click to view author's profile"
+            )
+              i.icon-button.fab.fa-github
```

Comment:
why are there some fix on the repo link as well?

Reply:
When you `merge group` **group by Author** the octocat icon links to the author's profile

Reply:
Change brought in PR #1017

Reply:
that's not the point, the point is that the scope of this PR has nothing to do with repo link

Reply:
Aah got it! I think its cause of the refactoring of code done along with it. Referring to @jamessspanggg comment on top.

Reply:
> Note that I also refactored the structure of the .summary-charts__title.
> 
> Previously, .summary-charts__title is separated into two, each for when filterGroupSelection is groupByAuthors or groupByRepos. There is overlapping code among the two, and adding another new icon/text to the title need to be done twice (once each for groupByAuthors and groupByRepos).
> 
> If this change is not needed or should be done in another PR, do let me know and I will act accordingly.

@eugenepeh Should I make the refactoring of code within this PR or another PR?

Reply:
its preferable to do it in a separate PR, given that 1 is a fix while the other is an added component on separate part of the code

Reply:
Noted. Putting this on hold until the refactoring is merged.